iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/twin/GetTwinTests.java

```diff
@@ -38,16 +47,47 @@ public GetTwinTests(IotHubClientProtocol protocol, AuthenticationType authentica
         super(protocol, authenticationType, clientType, publicKeyCert, privateKey, x509Thumbprint);
     }
 
-    @Before
-    public void setUpNewDeviceAndModule() throws IOException, IotHubException, URISyntaxException, InterruptedException, ModuleClientException, GeneralSecurityException
+    @Test
+    @StandardTierHubOnlyTest
+    public void testGetDeviceTwin() throws IOException, InterruptedException, IotHubException, GeneralSecurityException, ModuleClientException, URISyntaxException
     {
         super.setUpNewDeviceAndModule();
+        super.testGetDeviceTwin();
     }
 
     @Test
     @StandardTierHubOnlyTest
-    public void testGetDeviceTwin() throws IOException, InterruptedException, IotHubException
+    public void testGetDeviceTwinWithProxy() throws IOException, InterruptedException, IotHubException, GeneralSecurityException, ModuleClientException, URISyntaxException
     {
-        super.testGetDeviceTwin();
+        if (testInstance.protocol != IotHubClientProtocol.MQTT || testInstance.authenticationType != AuthenticationType.SAS || testInstance.clientType != ClientType.DEVICE_CLIENT)
+        {
+            // This test doesn't really care about the device side protocol or authentication, so just run it once
+            // when the device is using MQTT with SAS auth
+            return;
+        }
+
+        super.setUpNewDeviceAndModule();
```

Comment:
So if we are using in both the tests, why are we removing @Before block?

Reply:
The @Before runs before any of the short circuiting we do at line 66 above. Basically, there is no need to setup for a test that doesn't need to run

Reply:
It helps to cut down on device identity creation, which has been a bottleneck lately in our builds